### ISSUE TITLE
Entry Dockerfile is a duplicate but no duplicate handling strategy has been set

### ISSUE THREAD

**Issue by pstobiecki-ocado** [palantir/gradle-docker#518]
## What happened?

With a Gradle configuration like this one:
```groovy
docker {
    //  [...]

    dependsOn bootJar
    files bootJar.archivePath, fileTree('src/main/docker')
    dockerfile file('src/main/docker/Dockerfile')
    //  [...]
}
```

I'm getting the following error (with Gradle 7.3):
```
* What went wrong:
Execution failed for task ':edge:dockerPrepare'.
> Entry Dockerfile is a duplicate but no duplicate handling strategy has been set. Please refer to https://docs.gradle.org/7.3/dsl/org.gradle.api.tasks.Copy.html#org.gradle.api.tasks.Copy:duplicatesStrategy for details.
```

because the Dockerfile wants to be copied twice - once as part of the `files` config and then as itself. Providing a correct `duplicatesStrategy` for `dockerPrepare` could solve it, but it's impossible to do it without modifying the plugin code.

## What did you want to happen?

It would be great if:
* the `dockerPrepare` task provided a `duplicatesStrategy` that allows such a build to pass, or
* it was possible to configure a `duplicatesStrategy` for this task per build.

I worked around it by changing the Gradle config like this:
```groovy
files bootJar.archivePath, fileTree('src/main/docker').matching { exclude 'Dockerfile' }
```
but I'd much prefer for it to work out of the box.

**Comment by stuartraetaylor** [comment]
This still seems to be an issue.

I'm working around it in a multi-module project like this,

In the root build.gradle:
```
subprojects {
    pluginManager.withPlugin('com.palantir.docker') {
        dockerPrepare {
            // Workaround for https://github.com/palantir/gradle-docker/issues/518
            duplicatesStrategy = DuplicatesStrategy.INCLUDE
        }
    }
}
```

**Comment by lucasmfraser** [comment]
I don't think this is an issue with this plugin, but rather a caveat of Gradle 7. See these two threads:

- https://github.com/gradle/gradle/issues/17236
- https://discuss.gradle.org/t/gradle-7-fail-for-duplicates-in-copy-specs-has-no-duplicates-in-project/39834/

**Comment by GeorgHoffmeyer** [comment]
I had a bit of the same issue. Problem is, that the dockerfile is in your `src/main/docker` folder. So it is first added with the `files` and than a second time with `dockerfile`

### Solution 1:
Move `Dockerfile` to an other location than `src/main/docker`

### Solution 2:
remove `src/main/docker` from `files`
